docs/add-ons/testing.md

```diff
@@ -2,29 +2,46 @@
+Once running, you'll need to run the task (Terminal -> Run Task) 'Start Hass.io', which will bootstrap Supervisor and Home Assistant.  You'll then be able to access the normal onboarding process via the Home Assistant instance at http://localhost:8123/
 
-The [Community Add-on][hassio-vagrant] repository create a vagrant based development system. This Vagrant virtual machine allows you to test and play with Supervisor and Home Assistant, and is a great environment for add-on developers
+The add-on(s) under development will be automatically found in the Local Add-ons repository.
+
+[hassio-addons]: https://github.com/home-assistant/hassio-addons
+[remote-containers]: https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers
+[hassio-addon-devcontainer]: https://github.com/issacg/hassio-addon-devcontainer
+
+## Remote development
+
+If you require access to phyiscal hardware or other resources that cannot be locally emulated (for example, serial ports), the next best option to develop add-ons is by adding them to the local add-on repository on a real device running Home Assistant. To access the local add-on repository on a remote device, install either the [Samba add-on] or [SSH add-on] and copy the add-on files to a subdirectory of `/addons`.
+
+Right now add-ons will work with images that are stored on Docker Hub (using `image` from add-on config). To ensure that the add-on is built locally and not fetched from an upstream respository, ensure that the `image` key is *not* present in your `config.json`.
 
 [Samba add-on]: https://www.home-assistant.io/addons/samba/
 [SSH add-on]: https://www.home-assistant.io/addons/ssh/
-[hassio-vagrant]: https://github.com/hassio-addons/hassio-vagrant
 
 ## Local build
 
-You can build and try the addon on your developer machine also. Move all addon stuff into a temp folder. If you use `FROM $BUILD_FROM` you need set a base image with build args. Normally you can use follow base images:
+If you don't want to use the devcontainer environment, you can still build add-ons locally with Docker.  The recommended method is to use the [official build tool][hassio-builder] to create the docker images.
+
+Assuming that your addon is in the folder `/path/to/addon` and your docker socket is at `/var/run/docker.sock`, you can build the addon for all supported architectures by running the following: `docker run --rm -ti --name hassio-builder --privileged -v /path/to/addon:/data -v /var/run/docker.sock:/var/run/docker.sock:ro homeassistant/amd64-builder -t /data --all --test -i my-test-addon-{arch} -d local`
+
+If you don't want to use the official build tool, you can still build with standalone Docker. If you use `FROM $BUILD_FROM` you'll need set a base image with build args. Normally you can use follow base images:
```

Comment:
Typo:

```suggestion
If you don't want to use the devcontainer environment, you can still build add-ons locally with Docker. The recommended method is to use the [official build tool][hassio-builder] to create the docker images.

Assuming that your addon is in the folder `/path/to/addon` and your docker socket is at `/var/run/docker.sock`, you can build the addon for all supported architectures by running the following: `docker run --rm -ti --name hassio-builder --privileged -v /path/to/addon:/data -v /var/run/docker.sock:/var/run/docker.sock:ro homeassistant/amd64-builder -t /data --all --test -i my-test-addon-{arch} -d local`

If you don't want to use the official build tool, you can still build with standalone Docker. If you use `FROM $BUILD_FROM` you'll need to set a base image with build args. Normally you can use follow base images:
```

Also maybe we should convert the long command lines into multiline commands in a fenced code block to make it readable.

Reply:
👍 Will do shortly

Reply:
The comment is not resolved, please split commands into multiline commands to keep them readable.

Reply:
Sorry, missed the `multiline` when I first read it...